core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/resources/org/openjdk/jmc/flightrecorder/rules/jdk/messages/internal/messages.properties

```diff
@@ -519,6 +520,8 @@ ManyRunningProcessesRule_TEXT_RECOMMENDATION=If this is a server environment, it
 ManyRunningProcessesRule_INFO_LIMIT=Competing processes limit
 ManyRunningProcessesRule_INFO_LIMIT_LONG=The number of simultaneous processes needed to trigger an info notice
 MetaspaceOomRuleFactory_RULE_NAME=Metaspace Out of Memory
+MetaspaceOomRuleFactory_TEXT_CAUSE=Java class metadata is allocated in native memory (metaspace). When the amount of native memory needed for a class metadata exceeds MaxMetaSpaceSize, a java.lang.OutOfMemoryError exception with a detail MetaSpace is thrown. The amount of metaspace that can be used for class metadata is limited by the parameter MaxMetaSpaceSize, which is specified on the command line.
+MetaspaceOomRuleFactory_TEXT_ACTION=If MaxMetaSpaceSize, has been set on the command-line, increase its value. MetaSpace is allocated from the same address spaces as the Java heap. Reducing the size of the Java heap will make more space available for MetaSpace.
```

Comment:
Not a blocker, but I think it'd be nice if we checked the flag events (if available). We can say both what the current value is, and if it was set on the command line adjust the text. Something like "MaxMetaspaceSize was set to {maxMetaspaceSize} on the command line, increasing this value may help resolve the error."